src/util.ts

```diff
@@ -140,3 +153,84 @@ export function isYarnUsed(existsSync = fs.existsSync): boolean {
 export function getPkgManagerName(isYarnUsed?: boolean): 'yarn' | 'npm' {
   return isYarnUsed ? 'yarn' : 'npm';
 }
+
+/**
+ * Create a directory if not exist
+ * If already exist check if contains any ts file.
+ * If it returns true we can expect that the directory
+ * exist and does not contain ts files.
+ */
+export async function createSrcDir(
+  dirpath: string,
+  options?: Options
+): Promise<boolean> {
+  try {
+    fs.mkdirSync(dirpath, { mode: 0o774 });
+  } catch (error) {
+    if (error.code === 'EEXIST') {
+      utilLog('Found existing src directory.', options);
+      if (await isAnyTsFileInDir(dirpath)) {
+        utilLog('src directory has already some Typescript files.', options);
+        return false;
+      } else {
+        return true;
+      }
+    }
+    utilLog(new Error('Creation of source directory aborted'), options);
+    return false;
+  }
+  return true;
+}
+
+async function isAnyTsFileInDir(
+  dirpath: string,
+  options?: Options
+): Promise<boolean> {
+  try {
+    const extension = '.ts';
+    const files: string[] = fs.readdirSync(dirpath);
+    if (files && files.length) {
+      const firstTsFileName = files.find(
+        (file: string) => path.extname(file).toLowerCase() === extension
+      );
+      if (firstTsFileName !== undefined) {
+        return true;
+      }
+    }
+    return false;
+  } catch (error) {
+    utilLog(error, options);
+    return false;
```

Comment:
Would it be safer (conservative) to return true here? Personally I think it would be better to let the error bubble to the caller. This seems like a simple boolean check, and it doesn't seem to have enough context to know what to do with an error.

Reply:
The aim of the try-catch here is : 

1) declutter createSrcDir() which is already in catch at this point;
2) catch readdirSync() exceptions.

#### 1) declutter createSrcDir()

If you think it is cleaner I could create an intermediate function, like:
```typescript
function createSrcDir(...){
    try {
        fs.mkdirSync(dirpath);
        return true;
     } catch (error) {
       if (error.code === 'EEXIST') {
          return srcAlreadyPresent(dirpath, options);
      }
    return false;
  }
}

function srcAlreadyPresent(...){
  try {
      return (await isAnyTsFileInDir(dirpath));
  } catch (error) {
    return false;
  }
}

function isAnyTsFileInDir() {
  // ... some logic here
}
```

#### 2) catch readdirSync exceptions.

For Example a directory exist but we don't owe the right permissions.
The solution at 1) will avoid the try-catch here, and we could safely bubble up the exceptions.
My though were that the installation of the template is an optional step so I tried to make as silent as I could, to avoid to block the process for an unwanted feature, for instance when adding gts to an existing project.
Please tell me if this make sense or if you prefer another behavior.
As a side note this is the case that I would like to address with furthers test as in ofrobots/inline-fixtures#1